### PR TITLE
Define Methods:

### PR DESCRIPTION
**Pull request by pnookala-px**
Define new GetRoutines.

### CODE REVIEW

File: k8s/core/nodes.go [hunk 1]
```diff
@@ -385,3 +399,133 @@ func (c *Client) DrainPodsFromNode(nodeName string, pods []corev1.Pod, timeout t
...
+			if strings.Contains(k, "kubernetes.io/os") {
+				if strings.EqualFold(v, "linux") {
+					linNodes.Items = append(linNodes.Items, n)
+				}
+				break // break from label, os label found
+			}
+		}
+	}
+	return &linNodes, nil
+}
```
Comment (Glitchfix) [inline k8s/core/nodes.go:450]: we can move these methods into one e.g.
```go
GetNodesByOS(platform string) (*corev1.NodeList, error) {
	...
	if strings.Contains(k, "kubernetes.io/os") {
				if strings.EqualFold(v, platform) {
					linNodes.Items = append(linNodes.Items, n)
				}
				break // break from label, os label found
			}
	...
}
```

File: k8s/core/nodes.go [hunk 2]
```diff
@@ -385,3 +399,133 @@ func (c *Client) DrainPodsFromNode(nodeName string, pods []corev1.Pod, timeout t
...
+					if readyErr == nil {
+						linNodes.Items = append(linNodes.Items, n)
+					}
+					break // Break from the labels loop, since os is found.
+				}
+			}
+		}
+	}
+	return &linNodes, nil
+}
```
Comment (Glitchfix) [inline k8s/core/nodes.go:506]: these two can be merged into one in a similar manner

**Review by Glitchfix — COMMENTED** [review]
minor changes suggested to reduce LOC

**Review by Glitchfix — APPROVED** [review]

**Comment by pnookala-px** [comment]
Define GetLinuxNodes, GetWindowsNodes and GetReadyNodes methods.

**Pull request merged**